packages/commons/src/router/errorHandlers.ts

```diff
@@ -31,3 +35,35 @@ export function zodiosValidationErrorToApiProblem(
       )
     );
 }
+
+export function errorsToApiProblemsMiddleware(
+  error: SyntaxError,
+  req: WithZodiosContext<express.Request, ExpressContext>,
+  res: Response,
+  _next: NextFunction
+): Response {
+  const ctx = fromAppContext(req.ctx);
+  ctx.logger.error(`Error in request: ${parseErrorMessage(error)}`);
+
+  if (error instanceof SyntaxError) {
+    return res
+      .status(constants.HTTP_STATUS_BAD_REQUEST)
+      .send(
+        makeApiProblem(
+          badRequestError("Invalid request body"),
+          () => constants.HTTP_STATUS_BAD_REQUEST,
+          ctx
+        )
+      );
+  }
+
+  return res
+    .status(constants.HTTP_STATUS_BAD_REQUEST)
+    .send(
+      makeApiProblem(
+        badRequestError("Unexpected error"),
+        () => constants.HTTP_STATUS_BAD_REQUEST,
+        fromAppContext(req.ctx)
+      )
+    );
```

Comment:
In this case, it can be any error, since it's not a SyntaxError (see comment above). I would fall back to a generic error sent as 500, since if we reach this point, it means that the error was unexpected and not handled before in other middlewares/routers

```suggestion
  return res
    .status(constants.HTTP_STATUS_INTERNAL_SERVER_ERROR)
    .send(
      makeApiProblem(
        badRequestError("Unexpected error"),
        () => constants.HTTP_STATUS_INTERNAL_SERVER_ERROR,
        ctx
      )
    );
```